### DISCUSSION TITLE
how to use generated custom classes?

### DISCUSSION THREAD

**Discussion by JeffBeltran** [muhammadsammy/tailwindcss-classnames#227] [Q&A] [answered]
Hey there, sorry if this is a silly question, what is the best way to use the generated classnames? i know im missing something but im pretty new to TS and im not even sure where to look

**Comment by muhammadsammy** [comment, marked as answer]
Hello, @JeffBeltran! 

If you used the CLI you should import _the generated_ file (and **NOT** the actual library) into your code in order to get the customized classnames, like this:

**DO:**
```ts
import { classnames } from "relative/path/to/generated/tailwindcss-classnames";
```
**DO NOT**
```ts
import { classnames } from "tailwindcss-classnames";
```
as you won't get the custom/generated classnames from your config, instead you will have only the default tailwindcss classes.

---
and then use the classnames as normal:

```ts
import { classnames } from "relative/path/to/generated/tailwindcss-classnames";

const styles = classnames(.....);
```

eg. if you generated the file in 'src/types/tailwindcss-classnames' and you want to use the generated file in 'src/index.ts', the code will be something like this:

```ts
// src/index.ts
import { classnames } from "./types/tailwindcss-classnames";

const styles = classnames(.....);
```

**Reply by JeffBeltran** [reply]
ah man that is silly simple, guess i was just tired at the end of the day. Thanks for answering the question. If i get stuck i'll bug you again but i suspect this should be good.

**Reply by JeffBeltran** [reply]
Hey @muhammadsammy this worked (as expected) but curious if it's my setup or something i need to tweak but the auto complete feature vscode gave me with the default package stopped working when i switched to this method, is this due to a size issue? (running JIT mode)

**Reply by muhammadsammy** [reply]
It should work just the same as the default library. 
I'm not sure why it's not working for you so could you share your TailwindCSS config file (or a sample repo) so I can check against it?

**Reply by muhammadsammy** [reply]
Also the JIT compiler features are not fully supported. See https://github.com/muhammadsammy/tailwindcss-classnames/issues/204

**Reply by JeffBeltran** [reply]
yeah i'll see about getting something up, just curious if it could be related to size and vscode choking on it? i ended up getting the tailwind vscode plugin to work with the auto complete so i'm good from my end. Thanks again!

**Reply by ChristopherML** [reply]
Hey @muhammadsammy, Thank you so much for the library, great work! 

Could I suggest that a note addressing this issue, how to import so as to use the custom types, is added to the readme in the "Using the CLI" section? I think it would help with clarity for new users. Cheers. Thanks again!

**Reply by andykenward** [reply]
What about adding it to your projects `tsconfig.json` `compilerOptions` `paths` instead?

```json
{
  "compilerOptions": {
    "paths": {
      "tailwindcss-classnames": ["__generated__/tailwindcss-classnames.ts"]
    }
  },
}
```

Which should allow you keep the import the same and use your generated version of `tailwindcss-classnames` instead of the `node_modules` one.
```
import { classnames } from "tailwindcss-classnames"
```

**Reply by jamesarosen** [reply]
This would be great to have in the README!

**Reply by muhammadsammy** [reply]
Added to the readme. Thank you all!
